From be1ee8b41ce35e9a49b75362bbe18fad9669c526 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Tue, 18 Oct 2022 14:02:21 +0300
Subject: [PATCH] [SW-66326] gaudi3: add support to enable pcie access

This patch adds support to enable pcie access in
ppboot boot sequence which should be enabled post
hard reset.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: Ic0bbe0be0311b4a8790a9ce6fab84ff2f183a057
---

diff --git a/arch/gaudi3/common/common.c b/arch/gaudi3/common/common.c
index 64e238b..b5b026e 100644
--- a/arch/gaudi3/common/common.c
+++ b/arch/gaudi3/common/common.c
@@ -1,4 +1,28 @@
-// SPDX-License-Identifier:     GPL-2.0+
+// SPDX-License-Identifier: GPL-2.0+
 /*
  * Copyright (C) 2022 HabanaLabs Ltd.
  */
+
+#include <preboot.h>
+#include <hl.h>
+#include <d0_pcie_wrap_regs.h>
+#include <d0_pcie_core_regs.h>
+#include <d0_pcie_aux_regs.h>
+#include <d0_pcie_elbi_rr_mstr_if_rr_lbw_regs.h>
+
+void hl_pcie_enable_access(void)
+{
+	unsigned int rst_src;
+
+	/* FSM blocks all range hence openning here */
+	regmap_writel(0, mmD0_PCIE_ELBI_RR_MSTR_IF_RR_LBW_PRIV_RANGE_EN_SHORT_13);
+	rst_src = hl_rst_src_get();
+
+	if ((rst_src == HL_FLR_RST) || (rst_src == HL_PRSTN_RST))
+		regmap_writel(0, mmD0_PCIE_CORE_REQ_RETRY_EN);
+
+	/* Enable downstream access */
+	regmap_writel(0, mmD0_PCIE_AUX_HOST_TERM_CTRL);
+	/* Enable PCIe outbound memory access  */
+	regmap_writel(0, mmD0_PCIE_WRAP_MASK_REQ);
+}
